.github/workflows/main-workflow.yaml

```diff
@@ -0,0 +1,203 @@
+
+  create-build:
+    name: Create ${{ matrix.target }} build
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        target: [ iOS, Android ]
+        include:
+          - os: macos-latest
+            target: iOS
+            build_target: ios
+            build_args: --no-codesign
+            build_path: build/ios/iphoneos
+            asset_extension: .zip
+            asset_content_type: application/zip
+          - os: ubuntu-latest
+            target: Android
+            build_target: appbundle
+            build_args: ''
+            build_path: build/app/outputs/bundle/release
+            asset_extension: .aab
+            asset_content_type: application/zip
+      # Disable fail-fast as we want results from all even if one fails.
+      fail-fast: false
+    needs: [ validationWorkflow, tagValidation ]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: subosito/flutter-action@v1.5.0
+      - name: Install Android dependencies
+        if: matrix.target == 'Android'
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - run: flutter doctor -v
+      - name: Install iOS dependencies
+        if: matrix.target == 'iOS'
+        run: |
+          flutter pub get
+          cd ios
+          rm Podfile.lock
+          pod install --repo-update
+          flutter clean
+
+      # Get dependencies and decrypt needed files.
+      - run: flutter pub get
+
+      - name: Decrypt certificates files
+        run: |
+          chmod +x ./scripts/decrypt.sh
+          ./scripts/decrypt.sh
+        env:
+          ENCRYPTED_SIGNETS_API_CERT_PASSWORD: ${{ secrets.ENCRYPTED_SIGNETS_API_CERT_PASSWORD }}
+
+      # Build the application.
+      - name: Build the application
+        run: flutter build -v ${{ matrix.build_target }} ${{ matrix.build_args }} --release
+
+      - name: Rename Android build
+        if: matrix.target == 'Android'
+        run: mv app-release.aab $GITHUB_WORKSPACE/notre_dame_${{ matrix.target }}_${{ needs.tagValidation.outputs.version }}.aab
+        working-directory: ${{ matrix.build_path }}
+
+      - name: Compress iOS build
+        if: matrix.target == 'iOS'
+        run: |
+          mv Runner.app NotreDame.app
+          ditto -c -k --sequesterRsrc --keepParent NotreDame.app $GITHUB_WORKSPACE/notre_dame_${{ matrix.target }}_${{ needs.tagValidation.outputs.version }}.${{ matrix.asset_extension }}
+        working-directory: ${{ matrix.build_path }}
+
+      - name: Upload build artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.target }}
+          path: $GITHUB_WORKSPACE/notre_dame_${{ matrix.target }}_${{ needs.tagValidation.outputs.version }}.${{ matrix.asset_extension }}
+
+  create-draft-release:
+    name: Draft Github release
+    if: ${{ github.event_name == 'push' }}
+    needs: [ tagValidation, validationWorkflow, create-build ]
+    runs-on: ubuntu-20.04
```

Comment:
Is this version needed? You could use Ubuntu latest

Reply:
Yes, good catch!